hw/ip/otbn/util/otbn-rig

```diff
@@ -54,35 +47,117 @@ def main() -> int:
     # Write out the snippets to a JSON file
     ser_snippets = [snippet.to_json() for snippet in snippets]
     try:
-        with open(args.output, 'w') as out_file:
-            out_file.write(json.dumps(ser_snippets))
+        if args.output == '-':
+            json.dump(ser_snippets, sys.stdout)
+            # Add a newline at end of output: json.dump doesn't, and it makes a
+            # bit of a mess of some consoles.
+            sys.stdout.write('\n')
+        else:
+            with open(args.output, 'w') as out_file:
+                json.dump(ser_snippets, out_file)
     except OSError as err:
         sys.stderr.write('Failed to open json output file {!r}: {}.'
                          .format(args.output, err))
         return 1
 
-    # If assembly output was requested, dump that here.
-    if args.asm_output is not None:
+    return 0
+
+
+def asm_main(args: argparse.Namespace) -> int:
+    '''Entry point for the asm subcommand'''
+
+    insns_file = get_insns_file()
+    if insns_file is None:
+        return 1
+
+    # Load snippets JSON
+    try:
+        with open(args.snippets, 'r') as snippets_file:
+            snippets_json = json.load(snippets_file)
+    except OSError as err:
+        sys.stderr.write('Failed to open snippets file {!r}: {}.'
```

Comment:
For `sys.stderr.write()` it would be nice to add a newline at the end, otherwise output looks like this:

```
philipp@host ~/src/opentitan (rig-cmdline)
$ hw/ip/otbn/util/otbn-rig gen --size 10 | hw/ip/otbn/util/otbn-rig asm -o gen -
Failed to open snippets file '-': [Errno 2] No such file or directory: '-'.philipp@host ~/src/opentitan (rig-cmdline)
```

I typically prefer `print("msg" file=sys.stderr)` to ensure there's a newline.

Reply:
Oops, good catch. And I hadn't seen that `print` had a `file` argument: thanks!